pkg/adapter/incoming.go

```diff
@@ -95,20 +188,65 @@ func (l *listener) processIncoming(targetRepo *v1alpha1.Repository) (provider.In
 	l.event.Organization = org
 	l.event.Repository = repo
 
+	var isProviderTypeNotSet bool
+	if targetRepo.Spec.GitProvider == nil {
+		isProviderTypeNotSet = true
+	} else if targetRepo.Spec.GitProvider.Type == "" {
+		isProviderTypeNotSet = true
+	}
+
 	var provider provider.Interface
-	switch targetRepo.Spec.GitProvider.Type {
-	case "github":
+	if isProviderTypeNotSet {
 		provider = github.New()
-	case "gitlab":
-		provider = &gitlab.Provider{}
-	case "gitea":
-		provider = &gitea.Provider{}
-	case "bitbucket-cloud":
-		provider = &bitbucketcloud.Provider{}
-	case "bitbucket-server":
-		provider = &bitbucketserver.Provider{}
-	default:
-		return l.processRes(false, nil, l.logger, "", fmt.Errorf("no supported Git provider has been detected"))
+	} else {
+		switch targetRepo.Spec.GitProvider.Type {
+		case "github":
+			provider = github.New()
+		case "gitlab":
+			provider = &gitlab.Provider{}
+		case "gitea":
+			provider = &gitea.Provider{}
+		case "bitbucket-cloud":
+			provider = &bitbucketcloud.Provider{}
+		case "bitbucket-server":
+			provider = &bitbucketserver.Provider{}
```

Comment:
I think it is better to have a custom GitProvider type from string and consuming it across.
keeping raw string may lead maintainability issue in long term

Reply:
Yes agree

With PAC we are supporting only those VCS thats why hard coded as of now
But will consider this point if there is anything newly added